### ISSUE TITLE
Suggestion: People should be able to delete comments from their own diary entry, and turn off commenting

### ISSUE THREAD

**Issue by rory** [openstreetmap/openstreetmap-website#1989]
Hi all, here's something I suggest, and I wonder what other people think about it before I go off and make a patch:

"People should be able to delete comments from their own diary entry, and turn off commenting"

Should I work on that patch? Should this be implemented at all? Currently nearly all the comments on diary entries are fine, but I worry about continuing the "anyone can comment on anything" current system. We're leaving the OSM diary system open to spam, to harassment, to abuse. Let's close that loophole. "Don't read the comments" is a common quote for a reason. In my opinion, the diary belongs to that user account, it's their space, their blog post. They should be allowed to control what happens on that page. They should have full control to get rid of any comment they don't want.  Anyone who can comment can also write their own diary entry, and refer to the original post. So there is always a way to "reply" to something in the OSM diary system.

**Comment by tomhughes** [comment]
I might accept allowing comments to be turned off but allowing people to delete comments seems to be asking for trouble - it means somebody can post a proposal, delete any comments that disagree, and then it looks to all the world like it has consensus support.

If a comment is problematic then it should be reported.

**Comment by dieterdreist** [comment]
sent from a phone

> On 10. Sep 2018, at 14:44, Rory McCann <notifications@github.com> wrote:
> 
> In my opinion, the diary belongs to that user account, it's their space, their blog post. They should be allowed to control what happens on that page.


it’s on osmf servers and part of the community activity. They have the right to control what they write, but other people of the community should have the right to comment in place, i.e. where the reply belongs to.


> They should have full control to get rid of any comment they don't want.


I agree to this in the sense it would be ok if people could indicate whose comments they do not want to see (or whether to see comments at all), but they shouldn’t get the right to block comments for other people.

**Comment by rory** [comment]
> If a comment is problematic then it should be reported.

That's one solution for problematic comments, but a centralized moderation system then begs do the moderators know how to spot sexism, racism, homophobic dog whistles etc etc? My suggestion alleviates the burden (and responsibility) from those moderators, and spreads power around to individual users.

---

If comments could be deleted, but there was a message on the page saying "X comments have been removed", would that be acceptable? *(I'm not sure I like this idea)* Then it's clear that what's shown isn't the whole story?

Are diary entries used for proposals? IME that's mostly mailing list posts, or wiki pages. If comments can be deleted, then people won't trust a diary entry even more.

**Comment by simonpoole** [comment]
IMHO we already have a solution for this: post on you own blog.

I see it more or less as a condition of use of our infrastructure and profiting from the exposure associated with it, not really that different from editing OSM, that you are willing to enter in to discourse. 

Naturally we all know that the elephant in the room and the associated problem, really has nothing to do with allowing or not allowing comments.

**Comment by kamicut** [comment]
@woodpeck your suggestion is similar to how Github allows for [hiding and unhiding individual comments](https://help.github.com/articles/managing-disruptive-comments/). This would allow some degree of transparency without resorting to deleting comments outright.

**Comment by tomhughes** [comment]
The github case is different though because it's the repository owner that can hide them, not the person that opens a ticket.

**Comment by Zverik** [comment]
Rory, it seems you are implying that a blog entry author is always good and reasonable. What about more impulsive people, who write a controversial entry and then attack in comments anybody with a different opinion. You see such an entry, you spend half an hour to explain your opposing point of view — and then find your comment deleted. Would that feel good for you?

**Comment by Nakaner** [comment]
I think that we should wait until 6 to 12 months have passed since the introduction of the moderation feature in the Rails Port. After that time we could have a look

* how many comments have been reported,
* how many have been deleted and
* how many have been both reported and deleted

**Comment by matkoniecz** [comment]
> That's one solution for problematic comments, but a centralized moderation system then begs do the moderators know how to spot sexism, racism, homophobic dog whistles etc etc?

Typical centralized moderation system relies on people reporting cases that need an intervention. Moderators usually spend more time on processing reports than on looking for content violating community rules.

**Comment by rory** [comment]
@matkoniecz yes.. Why do you ask now? Has something happened that would be relevant? I haven't put any work into a patch yet. I would probably start with a patch to turn off future comments first.
